infra/modules/azure_app_service_plan_autoscaler/variables.tf

```diff
@@ -0,0 +1,157 @@
+variable "tags" {
+  type        = map(any)
+  description = "Resources tags"
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "Resource group to deploy resources to"
+}
+
+variable "target_service" {
+  type = object({
+    app_service_name  = optional(string)
+    function_app_name = optional(string)
+  })
+
+  validation {
+    condition = (
+      (var.target_service.app_service_name != null && var.target_service.function_app_name == null) ||
+      (var.target_service.app_service_name == null && var.target_service.function_app_name != null)
+    )
+    error_message = "Only one between \"app_service_name\" and \"function_app_name\" can have a value"
+  }
+}
+
+variable "scheduler" {
+  type = object({
+    high_load = optional(object({
+      start = object({
+        hour    = number
+        minutes = number
+      })
+      end = object({
+        hour    = number
+        minutes = number
+      })
+      name    = string
+      default = number
+      minimum = number
+    }), null)
+    low_load = optional(object({
+      start = object({
+        hour    = number
+        minutes = number
+      })
+      end = object({
+        hour    = number
+        minutes = number
+      })
+      name    = string
+      default = number
+      minimum = number
+    }), null)
+    normal_load = object({
+      default = number
+      minimum = number
+    })
+    maximum = optional(number, 30)
+  })
+
+  default = {
+    high_load = {
+      name = "high_load_profile"
```

Comment:
this format needs some documentation. for example, what's the difference between 'default' and 'minimum'? what `start` / `end` means?

Reply:
In Docusaurus or where else?

Reply:
here in the module itself (ie. in the variable description)